docs/writing-rules/rule-syntax.md

```diff
@@ -759,6 +760,74 @@ int("2147483646")
 
 This removes quotes (`'`, `"`, and `` ` ``) from both ends of the metavariable content. As a result, Semgrep detects `"2147483648"`, but it does **not** detect `"2147483646"`. This is useful when you expect strings to contain integer or float data.
 
+### `metavariable-name`
+
+The `metavariable-name` operator adds a constraint to the types of identifiers a metavariable is able to match. Currently the only constraint supported is on module or namespace an identifier originates from. This is useful for filtering results in languages which don't have a native syntax for fully qualified names, or languages where module names may contain characters which are not legal in identifiers (e.g., JavaScript, TypeScript). 
+
+:::info
+The `metavariable-name` operator requires the Pro engine, since it requires additional name resolution information. Moreover, name resolution information is higher fidelity with interfile enabled, and thus `metavariable-name` will be more accurate with interfile naming.
```

Comment:
```suggestion
The `metavariable-name` operator requires cross-file analysis, which you can enable following the steps in [Enable cross-file analysis](/docs/getting-started/cli#enable-cross-file-analysis). This is because it requires additional name resolution information. Moreover, name resolution information is more precise with cross-file enabled, and thus `metavariable-name` will be more accurate with cross-file naming.
```

Reply:
It does require Pro but not interfile (but works better with interfile). Do you have a preferred wording for the first sentence, given that?

Reply:
@kopecs in the **Semgrep AppSec Platform > Settings** tab, there is no setting in which we enable "only" Pro - the UI toggle enables both. Also, the toggle explicitly calls it **Cross-file** 😅 so I can't change the toggle name. 

Maybe we can say:

> The `metavariable-name` operator requires _advanced_ analysis, which you can enable following the steps in [Enable cross-file analysis](/docs/getting-started/cli#enable-cross-file-analysis) ...

Reply:
OK. Let me confirm what the default here is if that toggle is off. Not sure if we still have Pro intrafile or if it is OSS (or it depends on some other thing). We should disclaim it here either way, since CE users won't have access to it, but let me get back to you.

Reply:
Anything "Pro" branded is by default not OSS, so yeah, CE users will not have access to it. Are you ok with me pushing a commit to add a callout?

Reply:
Feel free to push whatever!